Sources/LiveViewNative/Utils/Font.swift

```diff
@@ -0,0 +1,358 @@
+            case .smallCaps:
+                base = base.smallCaps()
+            case .lowercaseSmallCaps:
+                base = base.lowercaseSmallCaps()
+            case .uppercaseSmallCaps:
+                base = base.uppercaseSmallCaps()
+            case .weight:
+                base = base.weight(try modifier.decode(Font.Weight.self, forKey: .properties))
+            case .width:
+                base = base.width(try modifier.decode(Font.Width.self, forKey: .properties))
+            case .leading:
+                base = base.leading(try modifier.decode(Font.Leading.self, forKey: .properties))
+            }
+        }
+        
+        self = base
+    }
+    
+    enum FontType: String, Decodable {
+        case system
+        case custom
+    }
+    
+    enum FontModifier: String, Decodable {
+        case bold
+        case italic
+        case monospaced
+        case monospacedDigit = "monospaced_digit"
+        case smallCaps = "small_caps"
+        case lowercaseSmallCaps = "lowercase_small_caps"
+        case uppercaseSmallCaps = "uppercase_small_caps"
+        case weight
+        case width
+        case leading
+    }
+    
+    enum CodingKeys: String, CodingKey {
+        case type
+        case properties
+        case modifiers
+        
+        enum Modifier: String, CodingKey {
+            case type
+            case properties
+        }
+        
+        enum System: String, CodingKey {
+            case style
+            case size
+            case design
+            case weight
+        }
+        
+        enum Custom: String, CodingKey {
+            case name
+            case size
+            case fixedSize = "fixed_size"
+            case style
+        }
+    }
+}
+
+/// A font's weight.
+///
+/// Possible values:
+/// * `ultra_light`
+/// * `thin`
+/// * `light`
+/// * `regular`
+/// * `medium`
+/// * `semibold`
+/// * `bold`
+/// * `heavy`
+/// * `black`
+///
+/// You can also use the numeric value for the associated weight, such as `400` for regular and `900` for `black`.
+#if swift(>=5.8)
+@_documentation(visibility: public)
+#endif
+extension Font.Weight: Decodable {
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.singleValueContainer()
+        if let name = try? container.decode(String.self) {
+            switch name {
+            case "ultra_light": self = .ultraLight
+            case "thin": self = .thin
+            case "light": self = .light
+            case "regular": self = .regular
+            case "medium": self = .medium
+            case "semibold": self = .semibold
+            case "bold": self = .bold
+            case "heavy": self = .heavy
+            case "black": self = .black
+            default: throw DecodingError.dataCorrupted(.init(codingPath: [], debugDescription: "Unknown weight named \(name)"))
+            }
+        } else {
+            switch try container.decode(Int.self) {
+            case 100:
```

Comment:
These values aren't defined by SwiftUI(?), so I'm not sure if we should permit them?

Reply:
Good point, removed.